editor/src/components/canvas/dom-walker.ts

```diff
@@ -287,19 +289,82 @@ interface RunDomWalkerParams {
   selectedViews: Array<ElementPath>
   scale: number
   additionalElementsToUpdate: Array<ElementPath>
+  elementsToFocusOn: ElementsToRerender
 
   domWalkerMutableState: DomWalkerMutableStateData
   rootMetadataInStateRef: { readonly current: readonly ElementInstanceMetadata[] }
 }
 
+function runSelectiveDomWalker(
+  elementsToFocusOn: Array<ElementPath>,
+  domWalkerMutableState: DomWalkerMutableStateData,
+  selectedViews: Array<ElementPath>,
+  scale: number,
+  rootMetadataInStateRef: { readonly current: readonly ElementInstanceMetadata[] },
+  containerRectLazy: () => CanvasRectangle,
+): { metadata: ElementInstanceMetadata[]; cachedPaths: ElementPath[] } {
+  let workingMetadata: ElementInstanceMetadata[] = []
+
+  const canvasRootContainer = document.getElementById(CanvasContainerID)
+  if (canvasRootContainer != null) {
+    const validPaths: Array<ElementPath> | null = optionalMap(
```

Comment:
You might find that putting these into a `Set` and later on using `has` will be quicker.

Reply:
done